pkg/network/operator_test.go

```diff
@@ -0,0 +1,299 @@
+
+		if testCase.expectedError == "" {
+			assert.Nil(t, err)
+			assert.Equal(t, clusterNetworkName, networkOperator.Name)
+		} else {
+			assert.EqualError(t, err, testCase.expectedError)
+		}
+	}
+}
+
+func TestOperatorExists(t *testing.T) {
+	testCases := []struct {
+		testBuilder *OperatorBuilder
+		exists      bool
+	}{
+		{
+			testBuilder: newOperatorBuilder(buildTestClientWithDummyNetworkOperator()),
+			exists:      true,
+		},
+		{
+			testBuilder: newOperatorBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			exists:      false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exists := testCase.testBuilder.Exists()
+		assert.Equal(t, testCase.exists, exists)
+	}
+}
+
+func TestOperatorUpdate(t *testing.T) {
+	testCases := []struct {
+		testBuilder   *OperatorBuilder
+		expectedError error
+	}{
+		{
+			testBuilder:   newOperatorBuilder(buildTestClientWithDummyNetworkOperator()),
+			expectedError: nil,
+		},
+		{
+			testBuilder:   newOperatorBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("network.operator object %s does not exist", clusterNetworkName),
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Nil(t, testCase.testBuilder.Definition.Spec.DeployKubeProxy)
+
+		testCase.testBuilder.Definition.ResourceVersion = "999"
+		testCase.testBuilder.Definition.Spec.DeployKubeProxy = ptr.To(true)
+
+		testBuilder, err := testCase.testBuilder.Update()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, true, *testBuilder.Definition.Spec.DeployKubeProxy)
+		}
+	}
+}
+
+func TestOperatorWaitUntilInCondition(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		inCondition   bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			inCondition:   true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			inCondition:   true,
+			expectedError: fmt.Errorf("network.operator object %s does not exist", clusterNetworkName),
+		},
+		{
+			exists:        true,
+			inCondition:   false,
+			expectedError: context.DeadlineExceeded,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.exists {
+			networkOperator := buildDummyNetworkOperator()
+
+			if testCase.inCondition {
+				networkOperator.Status.OperatorStatus.Conditions = []operatorv1.OperatorCondition{{
+					Type:   operatorv1.OperatorStatusTypeAvailable,
+					Status: operatorv1.ConditionTrue,
+				}}
+			}
+
+			runtimeObjects = append(runtimeObjects, networkOperator)
+		}
+
+		testSettings := clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects:  runtimeObjects,
+			SchemeAttachers: operatorTestSchemes,
+		})
+		testBuilder := newOperatorBuilder(testSettings)
+
+		err := testBuilder.WaitUntilInCondition(operatorv1.OperatorStatusTypeAvailable, time.Second, operatorv1.ConditionTrue)
+		assert.Equal(t, testCase.expectedError, err)
+	}
+}
+
```

Comment:
Please add test cases for:
SetLocalGWMode
SetMultiNetworkPolicy

Reply:
@kononovn I'm honestly not sure the best approach to testing these since we really can't cover the functions without modifying the conditions while the function is running. Should I start a goroutine to update the conditions?

Reply:
ok just ignore my comment.